### DISCUSSION TITLE
How and /best practices recommended to control the PS module installed path with one or two location (Core & Windows PS) and perform version upgrade every 3 months without impacting existing code consumed by developers in the repository?

### DISCUSSION THREAD

**Discussion by MHA1981** [PowerShell/PowerShell#17852] [Q&A]
I am pretty new to PowerShell. Somebody please recommend that what would be the best practice if I want to use a single or max of 2 module paths for all the module installations (both Core and Windows PowerShell) which is shared across the Organization? We are using multiple agent servers configured in DevOps tool (Octopus) and PS modules were installed in all the servers.  But it is quite messy at the moment since Developers can also install the PS module (currently no standard defined) while they deploy their code. And all the code deployment are running using Octopus service account which is an Administrator in above mentioned Agent servers. This is caused an uncontrolled environment with multiple module path of PowerShell and multiple version of same modules. So currently it is difficult for us to manage the environment if we need to upgrade the agent servers/pwsh installation, module version upgrade or tidy-up module installed path. And no track of who is using older version of PS module and not sure if we install newest module version will be impacted any of the existing developer code.

1) Can we control the module installation path , if yes how?
2) All the latest PS module versions compatible with older versions? If we upgrade all the existing version to newer this will impact existing developer code which have in case any direct reference to the older version? Or PS compatibility feature will take care of this?
3) If we move all the existing installation to one or two path (separate path for Windows and Core) have any impact to the existing developer code if they referenced the path with Scope (current user, All user, etc.) option? 
4) How do we set up a best practice, something like upgrade the existing versions with latest periodically?

**Comment by kasini3000** [comment]
i have a solution
Would you like to resubmit the issue at here？
https://github.com/kasini3000/kasini3000/issues/new/choose
Someone once complained that I was here to discuss your topic , and I don't know if it's allowed to be discussed here.
